### PR TITLE
Now supporting #pragma once as an header guard

### PR DESCRIPTION
**Pull request by lodi-g**
It shall be activated with the "epitech-c-cpp-headers.usePragmaOnce" settings field

Please not that is untested as for some reason I can't start the extension by running the build task (module vscode not found).

Thanks

### CODE REVIEW

**Comment by Hirevo** [comment]
Hi,

First of all, thank you for submitting to this repo, the extension is being used more than I expected and I didn't think I would ever get pull requests for it. So it's very nice to see that the extension is somewhat useful to people and that some of them, like you, are willing to contribute.

To be honest, I didn't knew about #pragma once as an header guard, I only ever used and known #ifndef, so it was a great discovery.
You successfully added it very simply so I'll be approving and merging it to the master branch and we could package it with automatic C++ class generation (a feature I'm working on) for the 1.8.0 release.

Big thanks again for this great addition ! :D

File: src/extension.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ interface Config {
     username: string;
     login: string;
     headerType: string;
+    usePragmaOnce: Boolean;
```
Comment (Hirevo) [inline src/extension.ts]: Just a little thing, "Boolean", if I'm correct, refers to the Javascript object and "boolean" refers to the Typescript type for it. So I might switch it to "boolean" as this is an interface declaration which is exclusive to Typescript.

**Comment by Hirevo** [comment]
Also, well done for the non-tested code, I cloned your forked project to test the feature and it worked pretty much out-of-the-box (put aside the "usePragmaOnce" config option in package.json).

**Comment by lodi-g** [comment]
Hi,

Thanks for accepting my PR, glad to contribute to something I literally use every day. As a student currently doing the C++ pool I can tell you that anyone using VSCode uses your extension as well.

Thanks for pointing out that I forgot to change the package.json, this was the first time I worked with extensions and I wasn't aware about this.

Lastly, I wanted to add the choice in VSCode menus and at the plugin setup but without being able to test it's a non trivial task. Would you put a CONTRIBUTE.md file explaining how to modify and run the extension for contributors?

And about the Boolean, you're right. It actually doesn't need a capitalized B. Thanks for pointing it out as well.

Have a great day

**Comment by Hirevo** [comment]
I'm also in the middle of the C++ pool, I did this for the C and C++ pools at first and being useful was important to me, so glad it can be so for others.

I'll be commiting the fix for the package.json alongside the choice entry at plugin setup, I already completed it so it's just a commit to push for me.

As for the launch issue, I ran into this as well. The working solution for me is to launch a debug task from VSCode itself.
I'll be commiting the .vscode directory to get them configured with the CONTRIBUTE.md as you suggested.

Have a great day.

**Pull request merged**